tech_specs/event_logging.md

```diff
@@ -0,0 +1,61 @@
+# InfluxDB Event Logging
+
+Historic state of controllers should be written to a time-series database (influx) for later display to users.
+
+
+## Decoupling
+
+In order to decouple individual controllers and the history database, logging is done through events.
+Controllers publish events with their latest state, without knowing who reads it.
+
+The history service subscribes to these events, and writes the raw data to the database in a sensible manner.
+
+This way the controller services are not aware of the history service implementation, and the history implementation is not aware of controller service implementation details.
+
+## Data Structure
+
+After a subscription is set, it will relay all incoming messages.
+
+When relaying, the data dict is flattened.
+The first part of the routing key is considered the controller name, and becomes the InfluxDB measurement name.
```

Comment:
I think we should be even more generic than controller here. Service or data source. Not all sources of data control something.

Reply:
Fixed